src/main/scala/mesosphere/marathon/MarathonConf.scala

```diff
@@ -307,5 +307,12 @@ trait MarathonConf
     noshort = true,
     hidden = true,
     default = Some(MesosHeartbeatMonitor.DEFAULT_HEARTBEAT_FAILURE_THRESHOLD))
+
+  lazy val drainingTime = opt[Long](
+    "draining_time",
```

Comment:
We called it drainingTime because in Mesos terms, at the point an instance is scheduled for maintenance, the agent enters in "draining" state (http://mesos.apache.org/documentation/latest/maintenance/). 

I agree with you that draining sound like marathon will "migrate tasks to not-in-maintenance hosts". But currently marathon doesn't use HTTP scheduler, so it cannot migrate tasks taking as a reference the inversed offers. I hope marathon will at some point migrate to the HTTP scheduler, which will make it easy to implement the migration of tasks to another hosts.

About make it configurable for task, I do agree that it could be a good idea, but I'm unsure if this should be part of the first iteration (I would like to avoid changes on the deployment object). So I guess maybe it would be a good idea to change drainingTime to defaultDrainingTime, and then apps can override it. WDYT?

About taking an step back, I would really appreciate your feedback on this. I think because we have the specific use case for us (https://github.com/alanbover/deathnode), we could be failing into looking the big picture of the whole thing.